lib/ansible/module_utils/basic.py

```diff
@@ -479,9 +479,11 @@ def tmpdir(self):
             if basedir is not None and not os.path.exists(basedir):
                 try:
                     os.makedirs(basedir, mode=0o700)
-                except (OSError, IOError) as e:
-                    self.warn("Unable to use %s as temporary directory, "
-                              "failing back to system: %s" % (basedir, to_native(e)))
+                except OSError as ex:
```

Comment:
Since you're touching many `except` blocks, it's probably worth mentioning that using contextual names for the local variable typically improves readability. Turning an ex to native reads more ambiguous than a `to_native(dir_creation_err)`.